src/main/scala/com.socrata.tileserver/services/TileService.scala

```diff
@@ -227,13 +228,40 @@ object TileService {
     val dis = rs.open(new DataInputStream(resp.inputStream(Long.MaxValue)))
 
     try {
-      val headers = MsgPack.unpack(dis, MsgPack.UNPACK_RAW_AS_STRING).asInstanceOf[Map[String, Any]]
-      val jsonHeaders = JObject(headers.mapValues(v => JString(v.toString)))
-      headers.asInt("geometry_index") match {
-        case geomIndex: Any if geomIndex < 0 =>
-          Failure(InvalidSoqlPackException(headers))
-        case geomIndex: Any =>
-          Success(jsonHeaders -> new FeatureJsonIterator(dis, geomIndex))
+      val soqlIter = new SoQLPackIterator(dis)
+      val jsonHeaders = JObject(soqlIter.headers.mapValues(v => JString(v.toString)))
+      if (soqlIter.geomIndex < 0) {
+        Failure(InvalidSoqlPackException(soqlIter.headers))
+      } else {
+        val colNames = soqlIter.schema.map(_._1).toArray
+        val featureJsonIter = soqlIter.map { case soqlRow =>
+          val geom: Geometry = soqlRow(soqlIter.geomIndex) match {
+            case SoQLMultiPolygon(mp) => mp
+            case SoQLPolygon(p)       => p
+            case SoQLPoint(pt)        => pt
+            case SoQLMultiPoint(mp)   => mp
+            case SoQLLine(l)          => l
+            case SoQLMultiLine(ml)    => ml
+            case x: SoQLValue         => throw new RuntimeException("Should not be seeing non-geom SoQL type" + x)
+          }
+          val props = (0 until soqlRow.size).filterNot(_ == soqlIter.geomIndex).map { i =>
```

Comment:
This seems like it could also be pushed into the library.  Not sure what syntax I'd use.  

Essentially we want to convert the Iterator to a Map, getting the internals of it, and ignoring geometry columns.